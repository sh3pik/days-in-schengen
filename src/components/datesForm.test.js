import { shallow } from 'enzyme'
import React from 'react'
import DatesForm from './datesForm'

describe('DateForms', () => {
  const wrapper = shallow(<DatesForm />)

  it('renders empty form', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('updates state with data from the fields', () => {
    const inputElement = wrapper.find('input')
    inputElement.first().simulate('change', { target: { value: '06-06-2018' } })
    inputElement.last().simulate('change', { target: { value: '06-07-2018' } })
    expect(wrapper.state()).toEqual({
      dates: [{ start: '06-06-2018', end: '06-07-2018' }],
      spendDays: 0
    })
  })

  it('submits the form', () => {
    const inputElement = wrapper.find('input')
    wrapper.setState({
      dates: [{ start: '08-06-2018', end: '08-07-2018' }],
      spendDays: 0
    })
    wrapper.find('form').simulate('submit', { preventDefault() {} })
    expect(wrapper.state()).toEqual({
      dates: [{ start: '08-06-2018', end: '08-07-2018' }],
      spendDays: 2
    })
  })

  it('shows the result', () => {
    const buttonElement = wrapper.find('button')
    wrapper.setState({
      spendDays: 2
    })
    buttonElement.simulate('click')
    expect(wrapper.text()).toMatch('2 days spend')
  })
})
