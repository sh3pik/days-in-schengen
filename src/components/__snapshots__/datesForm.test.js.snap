// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateForms renders empty form 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DatesForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          Write days in mm-dd-yyyy format
</p>,
        Array [
          <div>
            <label>
                        Day In:
            </label>
            <input
                        onChange={[Function]}
            />
            <label>
                        Day Out:
            </label>
            <input
                        onChange={[Function]}
            />
</div>,
        ],
        <button>
          Press me
</button>,
        <div />,
      ],
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Write days in mm-dd-yyyy format",
        },
        "ref": null,
        "rendered": "Write days in mm-dd-yyyy format",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label>
              Day In:
</label>,
            <input
              onChange={[Function]}
/>,
            <label>
              Day Out:
</label>,
            <input
              onChange={[Function]}
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Day In:",
            },
            "ref": null,
            "rendered": "Day In:",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "onChange": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Day Out:",
            },
            "ref": null,
            "rendered": "Day Out:",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "onChange": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Press me",
        },
        "ref": null,
        "rendered": "Press me",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
        },
        "ref": null,
        "rendered": "",
        "type": "div",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            Write days in mm-dd-yyyy format
</p>,
          Array [
            <div>
              <label>
                            Day In:
              </label>
              <input
                            onChange={[Function]}
              />
              <label>
                            Day Out:
              </label>
              <input
                            onChange={[Function]}
              />
</div>,
          ],
          <button>
            Press me
</button>,
          <div />,
        ],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Write days in mm-dd-yyyy format",
          },
          "ref": null,
          "rendered": "Write days in mm-dd-yyyy format",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label>
                Day In:
</label>,
              <input
                onChange={[Function]}
/>,
              <label>
                Day Out:
</label>,
              <input
                onChange={[Function]}
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Day In:",
              },
              "ref": null,
              "rendered": "Day In:",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "onChange": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Day Out:",
              },
              "ref": null,
              "rendered": "Day Out:",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "onChange": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Press me",
          },
          "ref": null,
          "rendered": "Press me",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": "div",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
